### PR TITLE
[#183708862] Use ruby 3.1.0

### PR DESCRIPTION
**Pull request by AP-Hunt**
What
----

Upgrades to Ruby 3.1 for our tests and other scripts.

How to review
-------------

Make sure it can go out through a dev env OK. It should be a no-op.

---

🚨⚠️ Please do not merge this pull request via the GitHub UI ⚠️🚨

### CODE REVIEW

File: manifests/cf-manifest/spec/scripts/rotate_vars_store_secrets_spec.rb [hunk 1]
```diff
@@ -269,7 +269,7 @@
     it "onlies delete _old secrets for secrets set to rotate" do
       vars_to_rotate = vars_store.keys.select { |k| k.end_with? "_one" }
 
-      rotated_vars_store = rotate(manifest, vars_store, vars_to_rotate: vars_to_rotate, delete: true)
```
Comment (risicle) [inline manifests/cf-manifest/spec/scripts/rotate_vars_store_secrets_spec.rb:272]: :confused: This is the only transformation I don't immediately understand.

Comment (risicle) [reply manifests/cf-manifest/spec/scripts/rotate_vars_store_secrets_spec.rb:272]: Wait.. does omitting the value just bind it to the same-named local variable in ruby 3?

**Comment by risicle** [comment]
Missed some containers?

```
$ grep -r 364e267f048a07d8b41f25369a99ab0991a1fecc .
./concourse/pipelines/monitor-remote.yml:          tag: 364e267f048a07d8b41f25369a99ab0991a1fecc
./concourse/pipelines/test-certificate-rotation.yml:        tag: 364e267f048a07d8b41f25369a99ab0991a1fecc
./concourse/pipelines/test-certificate-rotation.yml:        tag: 364e267f048a07d8b41f25369a99ab0991a1fecc
./concourse/pipelines/test-certificate-rotation.yml:        tag: 364e267f048a07d8b41f25369a99ab0991a1fecc
./concourse/pipelines/test-certificate-rotation.yml:        tag: 364e267f048a07d8b41f25369a99ab0991a1fecc
./concourse/pipelines/test-certificate-rotation.yml:      tag: 364e267f048a07d8b41f25369a99ab0991a1fecc
```

**Comment by risicle** [comment]
Note that this release will have the "double tap" problem when first deployed - the first run of `pipeline-lock` will fail because it doesn't pick up docker image changes until new jobs are started.

File: manifests/cf-manifest/spec/scripts/rotate_vars_store_secrets_spec.rb (hunk 1 above)
Comment (AP-Hunt) [reply manifests/cf-manifest/spec/scripts/rotate_vars_store_secrets_spec.rb:272]: I have no idea, honestly. I allowed Rubocop to fix thing itself and it looks like something it sorted for itself; `bundle exec rubocop -A`

**Review by risicle — APPROVED** [review]

**Pull request closed without merging**